jms/src/main/scala/akka/stream/alpakka/jms/JmsConnector.scala

```diff
@@ -48,36 +53,115 @@ private[jms] trait JmsConnector[S <: JmsSession] { this: GraphStageLogic =>
       case d => d
     }
 
-    materializer match {
-      case m: ActorMaterializer => m.system.dispatchers.lookup(dispatcher.dispatcher)
-      case x => throw new IllegalArgumentException(s"Stage only works with the ActorMaterializer, was: $x")
-    }
+    ActorMaterializerHelper.downcast(materializer).system.dispatchers.lookup(dispatcher.dispatcher)
   }
 
-  protected def initSessionAsync(executionContext: ExecutionContext): Unit = {
-    ec = executionContext
-    val sessions: Seq[Future[S]] = openSessions()
-    val allSessions = Future.sequence(sessions)
+  protected def createSession(connection: jms.Connection, createDestination: jms.Session => jms.Destination): JmsSession
+
+  sealed trait ConnectionStatus
+  case object Connecting extends ConnectionStatus
+  case object Connected extends ConnectionStatus
+  case object TimedOut extends ConnectionStatus
+
+  protected def initSessionAsync(withReconnect: Boolean = true): Unit = {
```

Comment:
`withReconnect` is set to `false` for the publisher at this point, until we can be certain reconnects behaving correctly with the publisher. Then even the `if` block below can be taken out.